packages/relay/src/lib/services/cacheService/cacheService.ts

```diff
@@ -0,0 +1,213 @@
+ * limitations under the License.
+ *
+ */
+
+import { Logger } from 'pino';
+import { Registry } from 'prom-client';
+import { ICacheClient } from '../../clients/cache/ICacheClient';
+import { LocalLRUCache, RedisCache } from '../../clients';
+import { RedisCacheError } from '../../errors/RedisCacheError';
+
+/**
+ * A service that manages caching using different cache implementations based on configuration.
+ */
+export class CacheService {
+  /**
+   * The LRU cache used for caching items from requests.
+   *
+   * @private
+   */
+  private readonly internalCache: ICacheClient;
+
+  /**
+   * The Redis cache used for caching items from requests.
+   *
+   * @private
+   */
+  private readonly sharedCache: ICacheClient;
+
+  /**
+   * The logger used for logging all output from this class.
+   * @private
+   */
+  private readonly logger: Logger;
+
+  /**
+   * The metrics register used for metrics tracking.
+   * @private
+   */
+  private readonly register: Registry;
+
+  /**
+   * Used for reference to the state of REDIS_ENABLED and REDIS_URL env. variables.
+   */
+  private readonly isSharedCacheEnabled: boolean;
+  /**
+   * Represents a caching manager that utilizes various cache implementations based on configuration.
+   * @param {Logger} logger - The logger used for logging all output from this class.
+   * @param {Registry} register - The metrics register used for metrics tracking.
+   */
+  public constructor(logger: Logger, register: Registry) {
+    this.logger = logger;
+    this.register = register;
+
+    this.internalCache = new LocalLRUCache(logger.child({ name: 'localLRUCache' }), register);
+    this.sharedCache = this.internalCache;
+    this.isSharedCacheEnabled = this.isRedisEnabled();
+
+    if (this.isSharedCacheEnabled) {
+      this.sharedCache = new RedisCache(logger.child({ name: 'redisCache' }), register);
+    }
+  }
+
+  /**
+   * Checks whether Redis caching is enabled based on environment variables.
+   * @private
+   * @returns {boolean} Returns true if Redis caching is enabled, otherwise false.
+   */
+  private isRedisEnabled(): boolean {
+    const redisEnabled = process.env.REDIS_ENABLED && process.env.REDIS_ENABLED === 'true';
+    const redisUrlValid = process.env.REDIS_URL && process.env.REDIS_URL !== '';
+
+    if (redisEnabled && redisUrlValid) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Retrieves a value from the cache asynchronously.
+   *
+   * @param {string} key - The cache key.
+   * @param {string} callingMethod - The name of the calling method.
+   * @param {string} [requestIdPrefix] - The optional request ID prefix.
+   * @param {boolean} [shared=false] - Whether to use the shared cache (optional, default: false).
+   * @returns {Promise<any>} A Promise that resolves with the cached value or null if not found.
+   */
+  public async getAsync(
+    key: string,
+    callingMethod: string,
+    requestIdPrefix?: string,
+    shared: boolean = false
```

Comment:
Is shared needed now?
localLRUCache will never require `async` so should be served by the `get()` call.
I'd simplify and remove the use of `shared` and the else

Reply:
Removed it and the else. Thanks.